publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java

```diff
@@ -426,8 +426,7 @@ public Resource importResource(ResourceService resourceService, ImportSource imp
         return resourceService.importResource(this, importSource);
     }
 
-    public Resource updateResourceFromImport(ResourceService resourceService, ImportSource importSource) {
-        var userInstance = UserInstance.fromPublication(this.toPublication());
+    public Resource updateResourceFromImport(ResourceService resourceService, ImportSource importSource, UserInstance userInstance) {
         this.setResourceEvent(MergedResourceEvent.fromImportSource(importSource, userInstance, Instant.now()));
```

Comment:
This is the main change. UserInstance.fromPublication() serves us institution from existing publication, cause this is the value which is never updated for publication.